src/java.base/share/classes/sun/security/rsa/RSACore.java

```diff
@@ -434,20 +436,27 @@ boolean isReusable() {
     private static BlindingRandomPair getBlindingRandomPair(
             BigInteger e, BigInteger d, BigInteger n) {
 
-        ConcurrentLinkedQueue<BlindingParameters> queue = blindingCache.get(n);
+        ConcurrentLinkedQueue<BlindingParameters> queue;
+
+        lock.lock();
+        try {
+            queue = blindingCache.get(n);
```

Comment:
```suggestion
            queue = blindingCache.computeIfAbsent(n, ignored -> new ConcurrentLinkedQueue<>());
```

Reply:
Yeah, that makes sense after all the changes